apps/meteor/tests/unit/app/livechat/server/lib/sendTranscript.spec.ts

```diff
@@ -0,0 +1,191 @@
+			rid: 'rid',
+			token: 'token',
+			email: 'email',
+			subject: 'A custom subject',
+			user: { _id: 'x', name: 'x', utcOffset: '-6', username: 'x' },
+		});
+
+		expect(getTimezoneMock.calledWith({ _id: 'x', name: 'x', utcOffset: '-6', username: 'x' })).to.be.true;
+		expect(modelsMock.Messages.findLivechatClosingMessage.calledWith('rid', { projection: { ts: 1 } })).to.be.true;
+		expect(modelsMock.Messages.findVisibleByRoomIdNotContainingTypesBeforeTs.called).to.be.true;
+		expect(
+			mailerMock.calledWith({
+				to: 'email',
+				from: 'test@rocket.chat',
+				subject: 'A custom subject',
+				replyTo: 'test@rocket.chat',
+				html: '<div> <hr></div>',
+			}),
+		).to.be.true;
+	});
+	it('should use subject from setting when no subject provided', async () => {
+		modelsMock.LivechatVisitors.getVisitorByToken.returns({ language: null });
+		modelsMock.LivechatRooms.findOneById.returns({ t: 'l', v: { token: 'token' } });
+		modelsMock.Messages.findVisibleByRoomIdNotContainingTypesBeforeTs.returns([]);
+		mockSettingValues.Livechat_transcript_email_subject = 'A custom subject obtained from setting.get';
+
+		await sendTranscript({
+			rid: 'rid',
+			token: 'token',
+			email: 'email',
+			user: { _id: 'x', name: 'x', utcOffset: '-6', username: 'x' },
+		});
+
+		expect(getTimezoneMock.calledWith({ _id: 'x', name: 'x', utcOffset: '-6', username: 'x' })).to.be.true;
+		expect(modelsMock.Messages.findLivechatClosingMessage.calledWith('rid', { projection: { ts: 1 } })).to.be.true;
+		expect(modelsMock.Messages.findVisibleByRoomIdNotContainingTypesBeforeTs.called).to.be.true;
+		expect(
+			mailerMock.calledWith({
+				to: 'email',
+				from: 'test@rocket.chat',
+				subject: 'A custom subject obtained from setting.get',
+				replyTo: 'test@rocket.chat',
+				html: '<div> <hr></div>',
+			}),
+		).to.be.true;
+	});
+	it('should fail if room provided is invalid', async () => {
+		modelsMock.LivechatRooms.findOneById.returns(null);
+		modelsMock.LivechatVisitors.getVisitorByToken.returns({ language: null });
+
+		await expect(sendTranscript({ rid: 'rid', email: 'email', logger: mockLogger })).to.be.rejectedWith(Error);
+	});
+
+	it('should fail if room doesnt have valid information', async () => {
+		modelsMock.LivechatRooms.findOneById.returns({ t: 'c' });
+		modelsMock.LivechatVisitors.getVisitorByToken.returns({ language: null });
+
+		await expect(sendTranscript({ rid: 'rid', email: 'email' })).to.be.rejectedWith(Error);
+
+		modelsMock.LivechatRooms.findOneById.returns({ t: 'l' });
+		await expect(sendTranscript({ rid: 'rid', email: 'email' })).to.be.rejectedWith(Error);
+
+		modelsMock.LivechatRooms.findOneById.returns({ t: 'l', v: { otherProp: 'xxx' } });
+		await expect(sendTranscript({ rid: 'rid', email: 'email' })).to.be.rejectedWith(Error);
+
+		modelsMock.LivechatRooms.findOneById.returns({ t: 'l', v: { token: 'xxx' } });
+		await expect(sendTranscript({ rid: 'rid', email: 'email', token: 'xveasdf' })).to.be.rejectedWith(Error);
+	});
```

Comment:
I think it would be better to have one expectation for every it, that way we know what is wrong with the room information.

Reply:
hmmmmmmmmmmmmmmmmmm